prediction_market_agent/agents/safe_guard_agent/safe_guard.py

```diff
@@ -120,12 +121,31 @@ def validate_safe_transaction(
     do_message: bool,
     ignore_historical_transaction_ids: set[str] | None = None,
 ) -> ValidationConclusion:
-    api_keys = APIKeys()
-
     detailed_transaction_info = safe_api_utils.get_safe_detailed_transaction_info(
         transaction_id=transaction_id
     )
-    all_addresses_from_tx = extract_all_addresses_or_raise(detailed_transaction_info)
+    return validate_safe_transaction_obj(
+        detailed_transaction_info=detailed_transaction_info,
+        do_sign_or_execution=do_sign_or_execution,
+        do_reject=do_reject,
+        do_message=do_message,
+        ignore_historical_transaction_ids=ignore_historical_transaction_ids,
+    )
+
+
+@observe()
+def validate_safe_transaction_obj(
+    detailed_transaction_info: DetailedTransactionResponse,
```

Comment:
So we can inject data from crafted testing set, instead of ids of real transactions.